### DISCUSSION TITLE
Cloning virtual folders

### DISCUSSION THREAD

**Discussion by gruanlc** [drakkan/sftpgo#670] [Q&A] [answered]
Hi, 
I would copy/clone a virtual folder (using web GUI), but it seems this is not possible. Or have I missed something?
In case of users such a cloning button is available.

**Comment by drakkan** [comment, marked as answer]
Hi,

I could add this feature but when cloning an object I have to hide sensitive data such as credentials (otherwise we have to decrypt the credentials and send them to the browser as plain text and this is a security issue). So you have to enter the credentials again after cloning the folder. 

I think using a template should be better here, maybe I have just to add a button to import the users/folders from template directly instead of exporting them as JSON. The template feature was initially added to export users/folders to other SFTPGo instances

**Reply by gruanlc** [reply]
That sounds very good. And hiding/removing sensitive data is fair.

**Reply by drakkan** [reply]
> That sounds very good. And hiding/removing sensitive data is fair.

I started working on this, today I'm quite busy but I could push this change in the next days

**Reply by drakkan** [reply]
Clone is now gone, you can use templates for both users and folders, take a look [here](https://github.com/drakkan/sftpgo/commit/1c51239da88562d17723baa0b3874bd88753c7fe).

You can download a build artificat for your OS from [here](https://github.com/drakkan/sftpgo/actions/runs/1689014404) or use one of the edge Docker variants to test this feature, please let me know if you find any issues, thanks
